tests/plugins/tableselection/tablecellfocus.js

```diff
@@ -0,0 +1,286 @@
+		},
+
+		'test focus should be not moved outside the table and the editor after exec: \'cell -> Merge Cells\' and pressing the TAB key': function( editor ) {
+			var	action = 'cellMerge',
+				editorName = editor.name + action;
+
+			bender.editorBot.create( {
+				name: editorName,
+				config: {
+					plugins: 'tab, tabletools, tableselection'
+				}
+			}, function( bot ) {
+				var editor = bot.editor,
+					editable = editor.editable();
+
+				bot.setHtmlWithSelection( '<table border="1">' +
+					'<tbody>' +
+						'<tr>' +
+							'<td>1_1</td>' +
+							'<td>1_2</td>' +
+						'</tr>' +
+						'<tr>' +
+							'<td>[success!</td>' +
+							'<td>2_2]</td>' +
+						'</tr>' +
+						'<tr>' +
+							'<td>3_1</td>' +
+							'<td>3_2</td>' +
+						'</tr>' +
+					'</tbody>' +
+				'</table>' );
+
+				bot.execCommand( action );
+
+				editable.fire( 'keydown', new CKEDITOR.dom.event( {
+					keyCode: 9,
+					shiftKey: false
+				} ) );
+
+				var selection = editor.getSelection(),
+					isSelectionInTable = selection.isInTable( true );
+
+				assert.isTrue( isSelectionInTable, 'Selection was moved outside the table after invoke ' + action + ' command.' );
+			} );
+
+		},
+
+		'test focus should be not moved outside the table and the editor after exec: \'cell -> Merge Cells\' and pressing the SHIFT + TAB key': function( editor ) {
+			var	action = 'cellMerge',
+				editorName = editor.name + 'previous' + action;
+
+			bender.editorBot.create( {
+				name: editorName,
+				config: {
+					plugins: 'tab, tabletools, tableselection',
+					style: 'display: none'
+				}
+			}, function( bot ) {
+				var editor = bot.editor,
+					editable = editor.editable();
+
+				bot.setHtmlWithSelection( '<table border="1">' +
+					'<tbody>' +
+						'<tr>' +
+							'<td>1_1</td>' +
+							'<td>1_2</td>' +
+						'</tr>' +
+						'<tr>' +
+							'<td>[success!</td>' +
+							'<td>2_2]</td>' +
+						'</tr>' +
+						'<tr>' +
+							'<td>3_1</td>' +
+							'<td>3_2</td>' +
+						'</tr>' +
+					'</tbody>' +
+				'</table>' );
+
+				bot.execCommand( action );
+
+				editable.fire( 'keydown', new CKEDITOR.dom.event( {
+					keyCode: 9,
+					shiftKey: true
+				} ) );
+
+				var selection = editor.getSelection(),
+					isSelectionInTable = selection.isInTable( true );
+
+				assert.isTrue( isSelectionInTable, 'Selection was moved outside the table after invoke ' + action + ' command.' );
+			} );
+
+		}
+	};
+
+	function assertTable( action, editorType, shiftKey ) {
+		var editorName = 'tableselection-' + action + '-' + editorType;
+
+		bender.editorBot.create( {
+			name: editorName,
+			config: {
+				plugins: 'tab, tabletools, tableselection'
+			}
+		}, function( bot ) {
+			var editor = bot.editor,
+				editable = editor.editable();
+
+			bot.setHtmlWithSelection( '<table border="1">' +
+				'<tbody>' +
+					'<tr>' +
+						'<td>1_1</td>' +
+						'<td>1_2</td>' +
+					'</tr>' +
+					'<tr>' +
+						'<td>success!^</td>' +
```

Comment:
Probably it could be caused by the fact that you use a collapsed selection which is probably translated to the native one, not our `tableselection` one. To activate the latter you'd need to select the whole cell.

Reply:
The collapsed selection was not causing the problem because there was a problem with moving focus, not operations on the table. Already fixed.